server/src/main/java/org/apache/iotdb/db/mpp/plan/planner/MemoryDistributionCalculator.java

```diff
@@ -0,0 +1,615 @@
+    processConsumeAllChildrenAtTheSameTime(node);
+    return null;
+  }
+
+  @Override
+  public Void visitDeviceViewInto(DeviceViewIntoNode node, MemoryDistributionContext context) {
+    processConsumeAllChildrenAtTheSameTime(node);
+    return null;
+  }
+
+  @Override
+  public Void visitVerticallyConcat(VerticallyConcatNode node, MemoryDistributionContext context) {
+    processConsumeAllChildrenAtTheSameTime(node);
+    return null;
+  }
+
+  @Override
+  public Void visitSchemaQueryMerge(SchemaQueryMergeNode node, MemoryDistributionContext context) {
+    processConsumeChildrenOneByOneNode(node);
+    return null;
+  }
+
+  @Override
+  public Void visitSchemaQueryScan(SchemaQueryScanNode node, MemoryDistributionContext context) {
+    processConsumeAllChildrenAtTheSameTime(node);
+    return null;
+  }
+
+  @Override
+  public Void visitSchemaQueryOrderByHeat(
+      SchemaQueryOrderByHeatNode node, MemoryDistributionContext context) {
+    processConsumeAllChildrenAtTheSameTime(node);
+    return null;
+  }
+
+  @Override
+  public Void visitTimeSeriesSchemaScan(
+      TimeSeriesSchemaScanNode node, MemoryDistributionContext context) {
+    processConsumeAllChildrenAtTheSameTime(node);
+    return null;
+  }
+
+  @Override
+  public Void visitDevicesSchemaScan(
+      DevicesSchemaScanNode node, MemoryDistributionContext context) {
+    processConsumeAllChildrenAtTheSameTime(node);
+    return null;
+  }
+
+  @Override
+  public Void visitDevicesCount(DevicesCountNode node, MemoryDistributionContext context) {
+    processConsumeAllChildrenAtTheSameTime(node);
+    return null;
+  }
+
+  @Override
+  public Void visitTimeSeriesCount(TimeSeriesCountNode node, MemoryDistributionContext context) {
+    processConsumeAllChildrenAtTheSameTime(node);
+    return null;
+  }
+
+  @Override
+  public Void visitLevelTimeSeriesCount(
+      LevelTimeSeriesCountNode node, MemoryDistributionContext context) {
+    processConsumeAllChildrenAtTheSameTime(node);
+    return null;
+  }
+
+  @Override
+  public Void visitCountMerge(CountSchemaMergeNode node, MemoryDistributionContext context) {
+    processConsumeChildrenOneByOneNode(node);
+    return null;
+  }
+
+  @Override
+  public Void visitCreateTimeSeries(CreateTimeSeriesNode node, MemoryDistributionContext context) {
+    processConsumeAllChildrenAtTheSameTime(node);
+    return null;
+  }
+
+  @Override
+  public Void visitSchemaFetchMerge(SchemaFetchMergeNode node, MemoryDistributionContext context) {
+    processConsumeChildrenOneByOneNode(node);
+    return null;
+  }
+
+  @Override
+  public Void visitSchemaFetchScan(SchemaFetchScanNode node, MemoryDistributionContext context) {
+    processConsumeAllChildrenAtTheSameTime(node);
+    return null;
+  }
+
+  @Override
+  public Void visitCreateAlignedTimeSeries(
+      CreateAlignedTimeSeriesNode node, MemoryDistributionContext context) {
+    processConsumeAllChildrenAtTheSameTime(node);
```

Comment:
all the write operation should throw exception

Reply:
done